src/vos/vos_obj.c

```diff
@@ -1036,8 +1036,9 @@ recx_iter_prepare(struct vos_obj_iter *oiter, daos_key_t *dkey,
 	if (rc != 0)
 		D_GOTO(failed, rc);
 
-	filter.fr_ex.ex_lo = 0;
-	filter.fr_ex.ex_hi = ~(0ULL);
+	filter.fr_ex.ex_lo = oiter->it_recx.rx_idx;
+	filter.fr_ex.ex_hi = oiter->it_recx.rx_nr == 0 ? ~(0ULL) :
```

Comment:
something like recx2filter(&filter, &oiter->it_recx) would save the duplication but minor